tests/unit/update-manager-test.js

```diff
@@ -198,4 +197,53 @@ describe('UpdateManager', () => {
 
   })
 
+  describe('editable', () => {
+    const self = {err: ''}
+    let updater, docuri, rterm, options, userCallback, loadStub
+    var loadStatus = 200
+
+    beforeEach(() => {
+      options = {}
+      userCallback = () => {}
+
+      updater = new UpdateManager()
+      // updater.store.fetcher.webOperation = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 1"})
+      // updater.store.fetcher.load = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 2"})
+
+    })
+
+    it('Should detect a document is editable from metadata', () => {
+      loadMeta(updater.store)
+      expect(updater.editable(doc1)).to.equal('SPARQL')
+    })
+
+    it('Should not detect a document is editable from metadata after flush', () => {
+      loadMeta(updater.store)
+      updater.flagAuthorizationMetadata()
+      expect(updater.editable(doc1)).to.equal(undefined)
+    })
+
+
+    it('Async version should detect a document is editable from metadata', () => {
+      loadMeta(updater.store)
+      updater.checkEditable(doc1).then(
+        res => expect(result).to.equal('SPARQL')
```

Comment:
The test will not wait for the promise to resolve and this expect will not be executed. You either need to `await` or pass in the `done` method to the test function and call it when all expects are done. (see)https://jestjs.io/docs/asynchronous

Reply:
Let's replace it with async/await as discussed

Reply:
```
    it('Async version should detect a document is editable from metadata', async () => {
      loadMeta(updater.store)
      expect(updater.editable(doc1)).to.equal('SPARQL')
      const res = await updater.checkEditable(doc1)
      expect(res).to.equal('SPARQL')
    })

    it('Async version should not detect a document is editable from metadata after flush', async () => {
      loadMeta(updater.store)
      await updater.flagAuthorizationMetadata()
      expect(updater.editable(doc1)).to.equal(undefined)

      const result = await updater.checkEditable(doc1)
      expect(result).to.equal('SPARQL')
    })
```